### ISSUE TITLE
IOOS RA query/utility package

### ISSUE THREAD

**Issue by MathewBiddle** [ioos/erddapy#257]
Similar to [gliderpy](https://github.com/ioos/gliderpy) and various other wrapper packages, we should think about an IOOS wrapper package for high level queries against the various IOOS ERDDAP endpoints. Maybe provide some high level utilities too?

A subset of RA endpoints are documented here [1]. However, it would probably be better to use the awesome-erddap json instead [2]. Or, identify another curated list.

[1] https://github.com/ioos/ioos-asset-inventory/blob/main/utils/ra_erddaps.json

[2] https://github.com/IrishMarineInstitute/awesome-erddap/blob/master/erddaps.json

**Comment by ocefpaf** [comment]
> I think we would probably want to wait until after the GSOC refactoring to work on this, but maybe it's a good code sprint project.

Indeed. However, we can start discussion this high level API now. For example, would something like this make sense?

```python
from ioos_erddap import ERDDAP

ds = ERDDAP(ra="secoora")
ds   ← list all dataset_ids

ds["dataset_id"] ← return a xarray object for that dataset using the OPeNDAP response to take advantage of the lazy loading.
```

and

```python
ds = ERDDAP(ra="secoora", **constraints)
ds   ← list all dataset_ids constrained in space/time/query string

ds["dataset_id"] ← return an in memory/fsspec xarray object for the constrained dataset
```

**Comment by ocefpaf** [comment]
@MathewBiddle do you believe it is worth adding this as a topic for the code sprint this year?

**Comment by MathewBiddle** [comment]
Yes!

**Comment by Harish-2003** [comment]
may i know what the code sprint is all about? and is this issue still open...?

**Comment by ocefpaf** [comment]
Hi @Harish-2003, you may find more information on the code sprint at https://ioos.github.io/ioos-code-sprint/

The short version is: The IOOS code sprint is where some of us sit together to try to solve a collection of selected issues and/or add features to our Software stack.

**Comment by mwengren** [comment]
Could we use the existing code base of [ioos/colocate](https://github.com/ioos/colocate) as a starting point?

**Comment by ocefpaf** [comment]
> Could we use the existing code base of [ioos/colocate](https://github.com/ioos/colocate) as a starting point?

This would be lower level. The goal would be for colocate to be a front-end for it.
